pkgs/tools/system/testdisk/default.nix

```diff
@@ -0,0 +1,74 @@
+{ mkDerivation
+, wrapQtAppsHook
+, stdenv
+, fetchurl
+, ncurses
+, libuuid
+, pkgconfig
+, enableJpeg ? false, libjpeg ? null
+, enableZlib ? false, zlib ? null
+, enableNtfs ? false, ntfs3g ? null
+, enableExtFs ? false, e2fsprogs ? null
+, enableEwf ? false, libewf ? null
+, enableQt ? false, qtbase ? null, qttools ? null, qwt ? null
+}:
+
+assert enableJpeg -> libjpeg != null;
+assert enableZlib -> zlib != null;
+assert enableNtfs -> ntfs3g != null;
+assert enableExtFs -> e2fsprogs != null;
+assert enableEwf -> libewf != null;
+assert enableQt -> qtbase != null;
+assert enableQt -> qttools != null;
+assert enableQt -> qwt != null;
+
+mkDerivation rec {
+  pname = "testdisk";
+  version = "7.1";
+  src = fetchurl {
+    url = "https://www.cgsecurity.org/testdisk-${version}.tar.bz2";
+    sha256 = "1zlh44w67py416hkvw6nrfmjickc2d43v51vcli5p374d5sw84ql";
+  };
+
+  enableParallelBuilding = true;
+
+  buildInputs = [
+    ncurses
+    libuuid
+    pkgconfig
+  ]
+  ++ stdenv.lib.optional enableJpeg libjpeg
+  ++ stdenv.lib.optional enableZlib zlib
+  ++ stdenv.lib.optional enableNtfs ntfs3g
+  ++ stdenv.lib.optional enableExtFs e2fsprogs
+  ++ stdenv.lib.optional enableEwf libewf
+  ++ stdenv.lib.optional enableQt [ qtbase qttools qwt ];
+
+  nativeBuildInputs = stdenv.lib.optional enableQt [ wrapQtAppsHook ];
```

Comment:
Isn't  wrapQtAppsHook only necessary when not using the qt-specific mkDerivation?

Reply:
Yep, turns out it's redundant.